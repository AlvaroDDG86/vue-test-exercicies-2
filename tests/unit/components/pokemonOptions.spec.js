import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions'
import { pokemonsMock } from '../mocks/pokemons'
describe('PokemonOptions', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons: pokemonsMock
            }
        })
    })
    test('Should match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })    

    test('should render 4 list, one for each pokemon', () => {
        const lies = wrapper.findAll('li')
        expect(lies.length).toBe(4)
        lies.forEach((li, index) => {
            expect(li.text()).toBe(pokemonsMock[index].name)
        })
    });

    test('should emmit on click element', () => {
        const lies = wrapper.findAll('li')
        lies.forEach((li) => {
            li.trigger('click')
        })

        // assert event count
        expect(wrapper.emitted('selectionPokemon').length).toBe(4)

        // assert event payload
        expect(wrapper.emitted('selectionPokemon')[0]).toEqual([1]) // deep
        expect(wrapper.emitted('selectionPokemon')[1]).toEqual([2]) // deep
        expect(wrapper.emitted('selectionPokemon')[2]).toEqual([3]) // deep
        expect(wrapper.emitted('selectionPokemon')[3]).toEqual([4]) // deep
    });
});