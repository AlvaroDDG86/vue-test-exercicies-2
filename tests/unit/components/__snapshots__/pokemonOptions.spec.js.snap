// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Should match snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>charmander</li>
    <li>venusaur</li>
  </ul>
</div>
`;
